WireDomain/Sources/WireDomain/Synchronization/OneOnOneResolver.swift

```diff
@@ -219,15 +220,37 @@ struct OneOnOneResolver: OneOnOneResolverProtocol {
 
     private func switchLocalConversationToMLS(
         mlsConversation: ZMConversation,
-        for user: ZMUser
+        for user: ZMUser,
+        userID: WireDataModel.QualifiedID
     ) async {
         await context.perform {
-            /// Move local messages from proteus conversation if it exists
-            if let proteusConversation = user.oneOnOneConversation {
-                /// Since ZMMessages only have a single conversation connected,
-                /// forming this union also removes the relationship to the proteus conversation.
+
+            // Note on proteus, it's possible to have 2 duplicate 1-1 conversations, so we need to fetch both
+            // conversations here.
+            let proteusConversations: [ZMConversation] = fetchAllTeamOneOnOneProteusConversations(
+                otherUserID: userID,
+                in: context
+            )
+
+            var allProteusConversations = Set(proteusConversations)
+            if let existingConversation = user.oneOnOneConversation,
+               existingConversation.messageProtocol == .proteus {
+                allProteusConversations.insert(existingConversation)
+            }
+
+            // move local messages from proteus conversations if they exist
+            for proteusConversation in allProteusConversations {
+                // Since ZMMessages only have a single conversation connected,
+                // forming this union also removes the relationship to the proteus conversation.
                 mlsConversation.mutableMessages.union(proteusConversation.allMessages)
-                mlsConversation.isForcedReadOnly = false
```

Comment:
Why did you remove `mlsConversation.isForcedReadOnly = false`?

Reply:
oups my bad I didn't realise that when porting the new code, fixed